pkg/cmd/pulumi/convert/convert.go

```diff
@@ -457,13 +482,153 @@ func runConvert(
+		}
+	}
+
+	if len(diagnostics) != 0 {
+		var errorDiags hcl.Diagnostics
+		for _, d := range diagnostics {
+			if d.Severity == hcl.DiagError {
+				errorDiags = append(errorDiags, d)
+			}
+		}
+
+		if len(errorDiags) != 0 {
+			return nil, diagnostics, nil
+		}
+	}
+
+	return allPackageDescriptors, nil, nil
+}
+
+func generateAndLinkSdksForPackages(
+	ctx context.Context,
+	ws pkgWorkspace.Context,
+	language string,
+	sdkTargetDirectory string,
+	convertOutputDirectory string,
+	pkgs map[string]*schema.PackageDescriptor,
+) error {
+	for _, pkg := range pkgs {
+		tempOut, err := os.MkdirTemp("", "gen-sdk-for-dependency-")
+		if err != nil {
+			return fmt.Errorf("failed to create temporary directory: %w", err)
+		}
+
+		var parameterizationValue []byte
+		if pkg.Parameterization != nil {
+			parameterizationValue = pkg.Parameterization.Value
+		} else {
+			// Only generate SDKs for packages that have parameterization for now, others should be implicit.
+			continue
+		}
+
+		pkgSchema, err := pcmd.SchemaFromSchemaSourceValueArgs(ctx, pkg.Name, parameterizationValue)
+		if err != nil {
+			return fmt.Errorf("creating package schema: %w", err)
+		}
+
+		err = pcmd.GenSDK(
+			language,
+			tempOut,
+			pkgSchema,
+			/*overlays*/ "",
+			/*local*/ true,
+		)
+		if err != nil {
+			return fmt.Errorf("error generating sdk: %w", err)
+		}
+
+		sdkOut := filepath.Join(sdkTargetDirectory, pkg.Parameterization.Name)
+		err = pcmd.CopyAll(sdkOut, filepath.Join(tempOut, language))
+		if err != nil {
+			return fmt.Errorf("failed to move SDK to project: %w", err)
+		}
+
+		err = os.RemoveAll(tempOut)
+		if err != nil {
+			return fmt.Errorf("could not remove temp dir: %w", err)
+		}
+
+		fmt.Printf("Generated local SDK for package '%s:%s'\n", pkg.Name, pkg.Parameterization.Name)
+
+		// Change working directory to the newly generated project output so
+		// workspace instance behaves.
+		startingDir, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("could not get working directory: %w", err)
+		}
+		err = os.Chdir(convertOutputDirectory)
+		if err != nil {
+			return fmt.Errorf("could not change to output directory: %w", err)
+		}
+		defer func() {
+			err = os.Chdir(startingDir)
+			if err != nil {
+				fmt.Fprintf(
+					os.Stderr,
+					"error returning to original working directory %q: %s",
+					startingDir,
+					err.Error(),
+				)
+			}
+		}()
+		_, _, err = ws.ReadProject()
+		if err != nil {
+			return fmt.Errorf("generated root is not a valid pulumi workspace %q: %w", convertOutputDirectory, err)
+		}
+
+		sdkRelPath := filepath.Join("sdks", pkg.Parameterization.Name)
+		err = pcmd.DoLocalSdkLinking(ws, language, "./", pkgSchema, sdkRelPath)
+		if err != nil {
+			return fmt.Errorf("failed to link SDK to project: %w", err)
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func errorf(subject hcl.Range, f string, args ...interface{}) *hcl.Diagnostic {
+	return diagf(hcl.DiagError, subject, f, args...)
+}
```

Comment:
Looks like there's only one use of this -- kill it? Think that means `diagf` can go too.